ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -1967,6 +1968,13 @@ private RelNode applyPreJoinOrderingTransforms(RelNode basePlan, RelMetadataProv
       generatePartialProgram(program, false, HepMatchOrder.DEPTH_FIRST,
           HiveExceptRewriteRule.INSTANCE);
 
+      // ?
+      if (conf.getBoolVar(ConfVars.HIVE_OPTIMIZE_REWRITE_COUNTDISTINCT_ENABLED)) {
+        generatePartialProgram(program, true, HepMatchOrder.TOP_DOWN,
+            new HiveRewriteCountDistinctToDataSketches(conf));
```

Comment:
As mentioned above, 1) let's use the general flag + specific one, and 2) let's not pass the full conf object.